### ISSUE TITLE
Setup failed for google_home: Requirements for google_home not found: ['glocaltokens==0.4.0']

### ISSUE THREAD

**Issue by vitru** [leikoilja/ha-google-home#315]
Updated both HA Core and Google Home HA integration and it can no longer boot due to 
Setup failed for google_home: Requirements for google_home not found: ['glocaltokens==0.4.0']

Tried to install it manually via pip inside the container but Alpine doesn't include 'cc', so it can't compile grpcio-tools package.

## Version of the integration

If you are unsure about the version check the [github release here](https://github.com/leikoilja/ha-google-home/releases).

```text
1.7.0
```

Home assistant version: core-2021.9.0.dev20210820

Logs

```text
bash-5.1# pip install glocaltokens==0.4.0
Collecting glocaltokens==0.4.0
  Using cached glocaltokens-0.4.0-py3-none-any.whl (36 kB)
Collecting zeroconf<0.30.0,>=0.28.8
  Using cached zeroconf-0.29.0-py3-none-any.whl (55 kB)
Collecting gpsoauth<2.0.0,>=1.0.0
  Downloading gpsoauth-1.0.0-py3-none-any.whl (6.4 kB)
Requirement already satisfied: grpcio==1.31.0 in /usr/local/lib/python3.9/site-packages (from glocaltokens==0.4.0) (1.31.0)
Requirement already satisfied: requests<3.0.0,>=2.25.1 in /usr/local/lib/python3.9/site-packages (from glocaltokens==0.4.0) (2.25.1)
Requirement already satisfied: simplejson<4.0.0,>=3.17.2 in /usr/local/lib/python3.9/site-packages (from glocaltokens==0.4.0) (3.17.3)
Collecting grpcio-tools==1.31.0
  Using cached grpcio-tools-1.31.0.tar.gz (2.1 MB)
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w3k_gg1z/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w3k_gg1z/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-qf1o69xp
         cwd: /tmp/pip-install-w3k_gg1z/grpcio-tools/
    Complete output (11 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w3k_gg1z/grpcio-tools/setup.py", line 112, in <module>
        if check_linker_need_libatomic():
      File "/tmp/pip-install-w3k_gg1z/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
        cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
      File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
        self._execute_child(args, executable, preexec_fn, close_fds,
      File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'cc'
```

**Comment by KapJI** [comment]
We should probably document this in readme. You need to install compiler (cc), installation process depends on your system.

In your case it's probably `apk install gcc` but I'm not sure.

**Comment by DurgNomis-drol** [comment]
There should be wheels build for it. I also get this when trying to install on my RPI 4. Before i just installed fine with out doing anything. I believe this started after 2021.8 or maybe 2021.7 😢

**Comment by KapJI** [comment]
We pinned version of grpcio, only newer versions come with wheels for arm.

**Comment by Sulexa** [comment]
Same here, I updated HA from 2021.4 to 2021.9 and I have the same error as Vitru.
I'm on a RPI4 running docker version of HA.
Do you have any solution ?

**Comment by Morphy99** [comment]
I believe this is the same issue I'm having with HA on docker ARM64. I followed [these ](https://github.com/leikoilja/ha-google-home/issues/288#issuecomment-886008075) instructions which worked for me.

**Comment by joostvkempen** [comment]
I'm having the same problem for some months now. Was hoping for some release with a fix, but didn't happen. Didn't have time to really sort it out. I'm running HA docker version on Synology NAS and with earlier HA versions it was working correctly.  Did some testing with disabling other custom components to rule out dependency conflicts. With all other integrations disabled, still not working. I can't remove this integration either. It seems stuck in some way. I have access to the command line of the container. I don't have access to the command line of the host machine (Synology NAS). I know you can use the command line on Synology with some workaround, however I don't want to do this as I'm dependent on this device for some other things.  Is it possible in some way to remove and reinstall this component by deleting some files or yaml-entries or something??

**Comment by Morphy99** [comment]
If you have access to cli in the container then you should be able to run the commands I linked to?

**Comment by joostvkempen** [comment]
> If you have access to cli in the container then you should be able to run the commands I linked to?

Ok, so I don't have to run these on the host machine? I'll try them!

**Comment by joostvkempen** [comment]
Do I understand this right, I have to do this every time the container is restarted?

**Comment by Morphy99** [comment]
> Do I understand this right, I have to do this every time the container is restarted?

No, just when you pull a new image (update)

**Comment by joostvkempen** [comment]
Tnx, seems to work!

**Comment by leikoilja** [comment]
Nice, happy to hear it worked out for you @joostvkempen. @Morphy99 thanks for the help. I guess we really need to add some lines to the README documentation with references to help ppl troubleshoot dependency issues like these :)

**Comment by Morphy99** [comment]
I can help with that if you like? It would be nice if we could these dependencies added to the ARM HA docker images tbh as it's a bit of a pain having to do this every time we update.

**Comment by leikoilja** [comment]
That would be amazing if you could add that info with references to README docs and open a pull request we could merge :) Thanks for offering the help! 🚀 
But also i do agree that opening another PR for HA docker images to include `gcc`/`cc` would be the ultimate solution

**Comment by joostvkempen** [comment]
Updated to HA 2022.2 this morning and this is not working anymore: the wheel takes very long building (+30 minutes) with high CPU, then I get some errors and a rollback on grpcio. When I restart the container after that, it fails. Device.config keeps loading forever and Home Assistant never ends starting. Only solution I found so far is recreating the container. So without Google Home integration.... 

I'll post some more logs later on. Anyone else experiencing this?

**Comment by KapJI** [comment]
It works for me just fine on x86 system. But I have some good news for you.
2022.2 finally upgraded grpcio version to 1.43.0 which includes wheels for arm devices and we can use it in glocaltokens. So you don't need to compile it anymore.

@leikoilja can you upgrade it in glocaltokens and update the documentation that compilation is no longer needed?

**Comment by joostvkempen** [comment]
In 2022.2.0 the integration is not working for me:

`2022-02-03 11:37:52 ERROR (SyncWorker_24) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-h4yq1ken/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-h4yq1ken/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-y39s96_8
cwd: /tmp/pip-install-h4yq1ken/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-h4yq1ken/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-h4yq1ken/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 11:37:55 WARNING (MainThread) [homeassistant.setup] Setup of default_config is taking over 10 seconds.
2022-02-03 11:38:08 ERROR (SyncWorker_8) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-tkt14iso/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-tkt14iso/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-jwurj5ym
cwd: /tmp/pip-install-tkt14iso/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-tkt14iso/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-tkt14iso/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 11:38:13 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: default_config
2022-02-03 11:38:23 ERROR (SyncWorker_18) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ujh40ggs/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ujh40ggs/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-qvb5cynh
cwd: /tmp/pip-install-ujh40ggs/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-ujh40ggs/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-ujh40ggs/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 11:38:23 ERROR (MainThread) [homeassistant.setup] Setup failed for google_home: Requirements for google_home not found: ['glocaltokens==0.5.2'].`

Home Assistant Dockerized on Synology DS720+ (Intel Gemini Lake)

**Comment by leikoilja** [comment]
Good find, @KapJI, thanks for communicating it 🚀  I ll add it to the TODOs and try picking up later these days :)

**Comment by leikoilja** [comment]
Should be solved in https://github.com/leikoilja/ha-google-home/releases/tag/v1.9.5
Thanks, @KapJI for the lead. 
@joostvkempen, @Morphy99, @vitru, @Sulexa please check that the latest release fixes your issue :)

**Comment by joostvkempen** [comment]
Sorry, issue still exists. Updated to 1.9.5 (using HACS), restarted HA, Invalid Config. Recreated HA Container and restart, still error.

Some of the logging I think is part of this:

```
2022-02-03 22:08:57 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration google_home which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-02-03 22:09:05 ERROR (Thread-4) [root] Uncaught thread exception
Traceback (most recent call last):
File "/usr/local/lib/python3.9/threading.py", line 973, in _bootstrap_inner
self.run()
File "/usr/local/lib/python3.9/threading.py", line 910, in run
self._target(*self._args, **self._kwargs)
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3591, in _thread_main
self.loop_forever(retry_first_connection=True)
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1756, in loop_forever
rc = self._loop(timeout)
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1164, in _loop
rc = self.loop_read()
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1556, in loop_read
rc = self._packet_read()
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2439, in _packet_read
rc = self._packet_handle()
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3033, in _packet_handle
return self._handle_publish()
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3327, in _handle_publish
self._handle_on_message(message)
File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3570, in _handle_on_message
on_message(self, self._userdata, message)
```
and

```
2022-02-03 22:09:35 ERROR (SyncWorker_17) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xvj5f_dt/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xvj5f_dt/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-jfkxqcb2
cwd: /tmp/pip-install-xvj5f_dt/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-xvj5f_dt/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-xvj5f_dt/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 22:09:50 ERROR (SyncWorker_7) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-084v5uwe/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-084v5uwe/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-sxgg6kek
cwd: /tmp/pip-install-084v5uwe/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-084v5uwe/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-084v5uwe/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 22:10:00 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: default_config
2022-02-03 22:10:06 ERROR (SyncWorker_2) [homeassistant.util.package] Unable to install package glocaltokens==0.5.2: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0r829bqn/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0r829bqn/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-_uxdag3r
cwd: /tmp/pip-install-0r829bqn/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-0r829bqn/grpcio-tools/setup.py", line 112, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-0r829bqn/grpcio-tools/setup.py", line 73, in check_linker_need_libatomic
cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cc'
----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
2022-02-03 22:10:06 ERROR (MainThread) [homeassistant.setup] Setup failed for google_home: Requirements for google_home not found: ['glocaltokens==0.5.2'].
2022-02-03 22:10:06 WARNING (MainThread) [homeassistant.helpers.frame] Detected integration that accessed discovery_info.__contains__('ssdp_location') instead of discovery_info.upnp.__contains__('ssdp_location') or discovery_info.ssdp_headers.__contains__('ssdp_location'); this will fail in version 2022.6. Please report issue for upnp using this method at homeassistant/components/upnp/device.py, line 78: if ssdp.ATTR_SSDP_LOCATION not in headers:
2022-02-03 22:10:06 WARNING (MainThread) [homeassistant.helpers.frame] Detected integration that accessed discovery_info['ssdp_location'] instead of discovery_info.ssdp_location, discovery_info.upnp['ssdp_location'] or discovery_info.ssdp_headers['ssdp_location']; this will fail in version 2022.6. Please report issue for upnp using this method at homeassistant/components/upnp/device.py, line 81: location = headers[ssdp.ATTR_SSDP_LOCATION]
2022-02-03 22:10:07 WARNING (SyncWorker_13) [pychromecast.dial] Failed to determine cast type
2022-02-03 22:10:07 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/discovery_flow.py", line 74, in _async_process_pending_flows
await gather_with_concurrency(
File "/usr/src/homeassistant/homeassistant/util/async_.py", line 185, in gather_with_concurrency
return await gather(
File "/usr/src/homeassistant/homeassistant/util/async_.py", line 183, in sem_task
return await task
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 203, in async_init
flow, result = await task
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 221, in _async_init
flow = await self.async_create_flow(handler, context=context, data=data)
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 730, in async_create_flow
await async_process_deps_reqs(self.hass, self._hass_config, integration)
File "/usr/src/homeassistant/homeassistant/setup.py", line 351, in async_process_deps_reqs
await requirements.async_get_integration_with_requirements(
File "/usr/src/homeassistant/homeassistant/requirements.py", line 83, in async_get_integration_with_requirements
await _async_process_integration(hass, integration, done)
File "/usr/src/homeassistant/homeassistant/requirements.py", line 99, in _async_process_integration
await async_process_requirements(
File "/usr/src/homeassistant/homeassistant/requirements.py", line 162, in async_process_requirements
await _async_process_requirements(
File "/usr/src/homeassistant/homeassistant/requirements.py", line 180, in _async_process_requirements
raise RequirementsNotFound(name, [req])
homeassistant.requirements.RequirementsNotFound: Requirements for google_home not found: ['glocaltokens==0.5.2'].
```

**Comment by KapJI** [comment]
It seems it can't install glocaltokens 0.5.2. Current version of the integration depends on 0.6.1. So you have the old version of the integration somewhere.

**Comment by leikoilja** [comment]
Ops, sorry gents, let's fix it up and get another patch version out :)  
https://github.com/leikoilja/ha-google-home/pull/452

**Comment by leikoilja** [comment]
Okay, fixed the missing reference and released 1.9.6 https://github.com/leikoilja/ha-google-home/releases/tag/v1.9.6.

However, I m on a Truenas setup myself and it's failing with:
```
022-02-04 08:31:37 ERROR (SyncWorker_0) [homeassistant.util.package] Unable to install package glocaltokens==0.6.1: ERROR: Command errored out with exit status 1:
command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-d2s801mr/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-d2s801mr/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-3h_hvxn_
cwd: /tmp/pip-install-d2s801mr/grpcio-tools/
Complete output (11 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-d2s801mr/grpcio-tools/setup.py", line 180, in <module>
if check_linker_need_libatomic():
File "/tmp/pip-install-d2s801mr/grpcio-tools/setup.py", line 91, in check_linker_need_libatomic
cpp_test = subprocess.Popen([cxx, '-x', 'c++', '-std=c++11', '-'],
File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__
self._execute_child(args, executable, preexec_fn, close_fds,
File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child
raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'c++'
```
as well. Even thou, i m on x86 system. Some more digging needs to be done :)

**Comment by joostvkempen** [comment]
Thanks for your quick responses and hard work! I'm no developer, so can't help you with that... I'm willing to test whenever necessary!

I've updated with 1.9.6 and upgraded HA to 2022.2.1 and failing:

```
Logger: homeassistant.util.package
Source: util/package.py:99
First occurred: 09:06:34 (3 occurrences)
Last logged: 09:07:02

Unable to install package glocaltokens==0.6.1: ERROR: Command errored out with exit status 1: command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ixd914ya/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ixd914ya/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-lccr82s6 cwd: /tmp/pip-install-ixd914ya/grpcio-tools/ Complete output (11 lines): Traceback (most recent call last): File "<string>", line 1, in <module> File "/tmp/pip-install-ixd914ya/grpcio-tools/setup.py", line 180, in <module> if check_linker_need_libatomic(): File "/tmp/pip-install-ixd914ya/grpcio-tools/setup.py", line 91, in check_linker_need_libatomic cpp_test = subprocess.Popen([cxx, '-x', 'c++', '-std=c++11', '-'], File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) FileNotFoundError: [Errno 2] No such file or directory: 'c++' ---------------------------------------- ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output. WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available. You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
Unable to install package glocaltokens==0.6.1: ERROR: Command errored out with exit status 1: command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-t83gqm2m/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-t83gqm2m/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-xvdv9eeb cwd: /tmp/pip-install-t83gqm2m/grpcio-tools/ Complete output (11 lines): Traceback (most recent call last): File "<string>", line 1, in <module> File "/tmp/pip-install-t83gqm2m/grpcio-tools/setup.py", line 180, in <module> if check_linker_need_libatomic(): File "/tmp/pip-install-t83gqm2m/grpcio-tools/setup.py", line 91, in check_linker_need_libatomic cpp_test = subprocess.Popen([cxx, '-x', 'c++', '-std=c++11', '-'], File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) FileNotFoundError: [Errno 2] No such file or directory: 'c++' ---------------------------------------- ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output. WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available. You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
Unable to install package glocaltokens==0.6.1: ERROR: Command errored out with exit status 1: command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-7kvcnu7w/grpcio-tools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-7kvcnu7w/grpcio-tools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-w0qh86xp cwd: /tmp/pip-install-7kvcnu7w/grpcio-tools/ Complete output (11 lines): Traceback (most recent call last): File "<string>", line 1, in <module> File "/tmp/pip-install-7kvcnu7w/grpcio-tools/setup.py", line 180, in <module> if check_linker_need_libatomic(): File "/tmp/pip-install-7kvcnu7w/grpcio-tools/setup.py", line 91, in check_linker_need_libatomic cpp_test = subprocess.Popen([cxx, '-x', 'c++', '-std=c++11', '-'], File "/usr/local/lib/python3.9/subprocess.py", line 951, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.9/subprocess.py", line 1821, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) FileNotFoundError: [Errno 2] No such file or directory: 'c++' ---------------------------------------- ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output. WARNING: You are using pip version 20.2.4; however, version 22.0.3 is available. You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
```

**Comment by KapJI** [comment]
Please continue the discussion of this issue in https://github.com/leikoilja/ha-google-home/issues/453
